nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ListFile.java

```diff
@@ -547,31 +549,67 @@ public boolean test(final Path path, final BasicFileAttributes attributes) {
             }
         };
 
-        final Stream<Path> inputStream = getPathStream(basePath, maxDepth, matcher);
+        try {
+            final long start = System.currentTimeMillis();
```

Comment:
The current solution is conceptually different than what is written in https://issues.apache.org/jira/browse/NIFI-7292.
This implementation skips inaccessible directories, even if they are not being filtered out.
Not necessarily a bad thing, but I think it should be clarified in the Jira if we want to accept this.
And/or add at least a test case with a PATH_FILTER.

Reply:
You are perfectly right with that there is a difference in the way my solution proposal was implemented. I tried to highlight this on the description but I did miss to directly notify the reporter of the ticket indeed. Now I asked for his approval so the future of the PR is highly depend on his answer.

As for the test I think the two mechanism in this implementation works somewhat independent but in order to properly cover the possibilities I added a case should cover the possible interference.